extensions/wikia/Search/classes/MediaWikiService.php

```diff
@@ -1080,14 +1080,33 @@ protected function getFileForPageId( $pageId ) {
 		return self::$pageIdsToFiles[$pageId];
 	}
 
+
+	/**
+	 * @param $pageIds
+	 */
+	public function initialPagesLoad( $pageIds ) {
+		$titles = \TitleBatch::newFromIds( $pageIds, DB_SLAVE );
+		/* @var \Title $title */
+		foreach ( $titles as $title ) {
+			$pageId = $title->getArticleID();
+			$article = new \Article( $title );
+			if ( $article === null ) {
```

Comment:
`TitleBatch::newFromIds()` does

```php
foreach ($res as $row) {
	$list[$row->page_id] = Title::newFromRow($row);
}
```
which filters out non-existing pages. Therefore `$titles` is expected to be a valid list of `Title`s for which non-`null` `Article`s will be returned.  I don't mind us to be extra cautious here, but instead of making the entire method `return null`, we should simply skip an item and `continue` here.

Reply:
`Article`s can be created from non-existing `Title`s.
```
michal@dev-michal:~$ SERVER_ID=177 php /usr/wikia/source/wiki/maintenance/eval.php
> $title = Title::newFromText('DoesNotExistObviously');
> var_dump($title->exists());
bool(false)
> $article = new Article( $title );
> var_dump( is_null( $article ) );
bool(false)
> var_dump( get_class( $article ) );
string(7) "Article"
```